scripts/src/questions_jsonlines_transformer.ts

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/*
+Usage: Convert answers with multiple parts into separate rows.
+
+ts-node server/setup/questions_jsonlines_transformer.ts \
+    --infile="./tmp/Questions.json" \
+    --out_scorings_file="./tmp/Scorings.json" \
+    --out_questions_file="./tmp/Questions2.json"
+
+bq --project=wikidetox --dataset_id=wikidetox:crowdsourcedb \
+  load \
+  --schema=./tmp/Scorings.schema.json \
+  --source_format=NEWLINE_DELIMITED_JSON \
+  crowdsourcedb.Scorings ./tmp/Scorings.json
+
+*/
+
+import * as byline from 'byline';
+import * as fs from 'fs';
+import * as yargs from 'yargs';
+
+import * as multiplex_transform from './multiplex_transform';
+
+export interface QuestionPartScores {
+  // Optional question: you get scores if you answer it, but you may skip.
+  optional?: boolean; // Is not set treated as false.
+
+  // Only one of these should be set.
+
+  // Associates a score value to each possible string answer.
+  enum?: { [enumValue: string]: number };
+
+  // Regexp where string gets score depending on match/no match.
+  stringRegExp?: {
+    regexp: string;
+    regexpFlags: string;
+    matchScore: number;
+    noMatchScore: number;
+  };
+  // Free test strings that always get the same score.
+  freeStringConstScore?: number;
+}
+
+// A questionaire (for a given question) consists of a named set of question parts.
+export interface QuestionScores {
+  [questionPartId: string]: QuestionPartScores;
+}
+
+// Command line arguments.
+interface Params {
+  infile: string;
+  out_questions_file: string;
+  out_scorings_file: string;
+}
+
+interface QuestionT1 {
+  accepted_answers: string;
+  question: string;
+  question_group_id: string;
+  question_id: string;
+  type: string;
+}
+
+type AnswerPartType =
+  | 'insult'
```

Comment:
Same question here about the leading '|'.

Also, any way of reusing this type across files? If you want to make this a lint only change, maybe just add a TODO so we can remember to do it later?

Reply:
w.r.t. `|` linter did it I think... will double check by removing and seeing if it adds it back. either way does compile. 

I'm in two minds, on the one hand, if it's actually a common type, then it being in a separate file for that kind of object makes sense. If on the other hand, one views it as a local definition, and each script is self contained, then it's better to repeat stuff.

Reply:
Yes, `gts fix` adds the leading `|`, which actually I like too. :)

Reply:
added TODO.

Reply:
I didn't know that syntax was acceptable--I guess I learned something new today! I do find it a bit weird, but who am I to object to auto-formatting tools? =)